### PR TITLE
Bootstrap JRuby using gradle

### PR DESCRIPTION
**Pull request by ph**
Instead of using rake to bootstrap we are slowly moving part of our
infra to gradle this is the first step taken, I didn't change the calls,
I just made the `rake bootstrap` call the `gradle bootstrap`

Closes: #8234

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,10 +27,61 @@ allprojects {
...
+      f.path = f.path.replaceFirst("^jruby-${jRubyVersion}", '')
+    }
+    exclude "**/stdlib/rdoc/**"
+    includeEmptyDirs = false
+    into "${projectDir}/vendor/jruby"
+}
+
+// If you are running a JRuby snapshot we will skip the integrity check.
+verifyFile.onlyIf { doChecksum }
+bootstrap.dependsOn downloadAndInstallJRuby
```
Comment (ph) [inline build.gradle]: TODO, have to check my vim syntax with gradle file.

**Comment by jordansissel** [comment]
Tested on windows:

* ./gradlew bootstrap
* jruby -S rake bootstrap

Both use gradle to download JRuby. Nice! One small step forward :heart:

File: build.gradle [hunk 2]
```diff
@@ -8,10 +27,61 @@ allprojects {
...
+
+// fetch version from Logstash's master versions.yml file
+def versionMap = (Map) (new Yaml()).load(new File("${projectDir}/versions.yml").text)
+
+String jRubyURL
+String jRubyVersion
+String jRubySha1
+Boolean doChecksum
+
+if (versionMap["jruby-runtime-override"]) {
```
Comment (jordansissel) [inline build.gradle]: I forgot about this feature; good idea porting this forward.

**Comment by ph** [comment]
I've changed the label, lets get that merged to master 6.1.

**Pull request closed without merging**